### PR TITLE
Updated the readFile function to correctly handle symbolic links

### PR DESCRIPTION
**Pull request by achmiel**
Fix to the `osquery::readFile` function (#1344).

### CODE REVIEW

**Comment by osqueryer** [comment]
Can one of the osquery admins reply with "ok to test" to kick off a build...

File: include/osquery/filesystem.h [hunk 1]
```diff
@@ -55,7 +55,7 @@ Status readFile(const boost::filesystem::path& path,
  *
  * @param path the path of the file that you would like to read.
  *
- * @return success iff the file would have been read. On success the status
```
Comment (theopolis) [inline include/osquery/filesystem.h]: Haha! So "iff" is slang for "if and only if", which is the intention here.

Comment (achmiel) [inline include/osquery/filesystem.h]: Oops, thanks, I'll fix that.

**Comment by marpaia** [comment]
@osqueryer ok to test

**Pull request merged**